test-data/unit/outputjson.test

```diff
@@ -32,10 +32,13 @@ def foo(x: int) -> None: ...
 def foo(x: Optional[int] = None) -> None:
     ...
 
+reveal_type(foo)
+
 foo('42')
 
 def bar() -> None: ...
 bar('42')
 [out]
-{"file": "main", "line": 12, "column": 0, "message": "No overload variant of \"foo\" matches argument type \"str\"", "hint": "Possible overload variants:\n    def foo() -> None\n    def foo(x: int) -> None", "code": "call-overload"}
-{"file": "main", "line": 15, "column": 0, "message": "Too many arguments for \"bar\"", "hint": null, "code": "call-arg"}
+{"file": "main", "line": 12, "column": 12, "message": "Revealed type is \"Overload(def (), def (x: builtins.int))\"", "hint": null, "code": "misc", "is_note": true}
```

Comment:
Could we change this to `"severity": "note"`, mirroring mypy's Python code (and severity error for the errors)? That will also make it easier to extend to other kinds of severity in the future.